lib/pf/node.pm

```diff
@@ -1327,6 +1331,32 @@ sub _cleanup_attributes {
     $info->{'status'} = _cleanup_status_value($info->{'status'});
 }
 
+=item node_extended_data
+
+Get the node extended data
+
+=cut
+
+sub node_extended_data {
+    my ($mac) = @_;
+    my @namespaces = qw(mse);
+    if (@{$Config{advanced}{extended_profile_filter_namespaces}}) {
+        my %hash;
+        my $redis = pf::Redis->new(server => 'localhost:6379');
+        my $json = JSON::MaybeXS->new;
+        foreach my $namespace (@namespaces) {
+            my ($data) = $redis->get("extended:${namespace}:${mac}");
```

Comment:
Wouldn't it be better to use the hashes in Redis instead of this to prevent race conditions while updating the JSON data.

Or maybe support both JSON in a key and hashes ?